<?xml version="1.0" ?>
<sdf version="1.5">
  <model name="satellite_dish">

    <link name="base">
      <inertial>
        <pose>0 -0.6077 0.3193 0 0 0</pose>
        <inertia>
          <ixx>165.1022</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>13.8377</iyy>
          <iyz>0</iyz>
          <izz>170.3359</izz>
        </inertia>
        <mass>400</mass>
      </inertial>
      <collision name="collision">
        <geometry>
          <mesh>
            <uri>model://satellite_dish/meshes/satellite_dish.dae</uri>
            <submesh>
              <name>Base</name>
            </submesh>
          </mesh>
        </geometry>
      </collision>
      <visual name="visual">
        <geometry>
          <mesh>
            <uri>model://satellite_dish/meshes/satellite_dish.dae</uri>
            <submesh>
              <name>Base</name>
            </submesh>
          </mesh>
        </geometry>
      </visual>
    </link>

    <link name="pivot">
      <pose>0 0 1.595 0 0 0</pose>
      <inertial>
        <inertia>
          <ixx>0.00073</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.000598</iyy>
          <iyz>0</iyz>
          <izz>0.00057</izz>
        </inertia>
        <mass>0.82</mass>
      </inertial>
      <collision name="collision">
        <pose>0 0 -1.595 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://satellite_dish/meshes/satellite_dish.dae</uri>
            <submesh>
              <name>Pivot</name>
            </submesh>
          </mesh>
        </geometry>
      </collision>
      <visual name="visual">
        <pose>0 0 -1.595 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://satellite_dish/meshes/satellite_dish.dae</uri>
            <submesh>
              <name>Pivot</name>
            </submesh>
          </mesh>
        </geometry>
      </visual>
    </link>

    <link name="dish">
      <pose>0 0 1.595 0.43 0 0</pose>
      <inertial>
        <pose>0 0.2 0 0 0 0</pose>
        <inertia>
          <ixx>9.2468</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>18.7729</iyy>
          <iyz>0</iyz>
          <izz>9.7826</izz>
        </inertia>
        <mass>170.056</mass>
      </inertial>
      <collision name="collision">
        <pose>0 -0.6519041 -1.451541 -0.43 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://satellite_dish/meshes/satellite_dish.dae</uri>
            <submesh>
              <name>Dish</name>
            </submesh>
          </mesh>
        </geometry>
      </collision>
      <visual name="visual">
        <pose>0 -0.6519041 -1.451541 -0.43 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://satellite_dish/meshes/satellite_dish.dae</uri>
            <submesh>
              <name>Dish</name>
            </submesh>
          </mesh>
        </geometry>
      </visual>
    </link>

    <link name="wheel_yaw">
      <pose>-0.2 -1.725 0.91 0.52 0 0</pose>
      <inertial>
        <inertia>
          <ixx>0.03698</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.03887</iyy>
          <iyz>0</iyz>
          <izz>0.07500</izz>
        </inertia>
        <mass>5.9697</mass>
      </inertial>
      <collision name="collision">
        <pose>0.2 1.065 -1.64 -0.52 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://satellite_dish/meshes/satellite_dish.dae</uri>
            <submesh>
              <name>Wheel_01</name>
            </submesh>
          </mesh>
        </geometry>
      </collision>
      <visual name="visual">
        <pose>0.2 1.065 -1.64 -0.52 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://satellite_dish/meshes/satellite_dish.dae</uri>
            <submesh>
              <name>Wheel_01</name>
            </submesh>
          </mesh>
        </geometry>
      </visual>
    </link>

    <link name="wheel_pitch">
      <pose>0.2 -1.725 0.91 0.52 0 0</pose>
      <inertial>
        <inertia>
          <ixx>0.03698</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.03887</iyy>
          <iyz>0</iyz>
          <izz>0.07500</izz>
        </inertia>
        <mass>5.9697</mass>
      </inertial>
      <collision name="collision">
        <pose>-0.2 1.065 -1.64 -0.52 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://satellite_dish/meshes/satellite_dish.dae</uri>
            <submesh>
              <name>Wheel_02</name>
            </submesh>
          </mesh>
        </geometry>
      </collision>
      <visual name="visual">
        <pose>-0.2 1.065 -1.64 -0.52 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://satellite_dish/meshes/satellite_dish.dae</uri>
            <submesh>
              <name>Wheel_02</name>
            </submesh>
          </mesh>
        </geometry>
      </visual>
    </link>

    <joint name="dish_yaw" type="revolute">
      <parent>base</parent>
      <child>pivot</child>
      <axis>
        <xyz>0 0 1</xyz>
        <dynamics>
          <damping>1000</damping>
        </dynamics>
      </axis>
    </joint>

    <joint name="dish_pitch" type="revolute">
      <parent>pivot</parent>
      <child>dish</child>
      <axis>
        <xyz>1 0 0</xyz>
        <limit>
          <lower>-0.2</lower>
          <upper>1.57</upper>
        </limit>
        <dynamics>
          <damping>100</damping>
        </dynamics>
      </axis>
    </joint>

    <joint name="wheel_yaw" type="revolute">
      <parent>base</parent>
      <child>wheel_yaw</child>
      <axis>
        <xyz>0 0 1</xyz>
      </axis>
    </joint>

    <joint name="wheel_pitch" type="revolute">
      <parent>base</parent>
      <child>wheel_pitch</child>
      <axis>
        <xyz>0 0 1</xyz>
      </axis>
    </joint>

    <plugin name="satellite_plugin" filename="libSatelliteDish.so">

      <yaw_ratio>0.1</yaw_ratio>
      <yaw_pid>1000 0 10</yaw_pid>

      <pitch_ratio>0.1</pitch_ratio>
      <pitch_pid>1000 100 10</pitch_pid>

      <frequency>100</frequency>
    </plugin>

  </model>
</sdf>
